Airwallex/AirwallexPayment/Sources/ApplePayProvider.swift

```diff
@@ -0,0 +1,208 @@
+            return true
+        } catch {
+            return false
+        }
+    }
+    
+    private let paymentController: PKPaymentAuthorizationController.Type
+    
+    init(delegate: any AWXProviderDelegate,
+         session: Session,
+         methodType: AWXPaymentMethodType?,
+         apiClient: AWXAPIClient = AWXAPIClient(configuration: .shared()),
+         paymentController: PKPaymentAuthorizationController.Type = PKPaymentAuthorizationController.self) {
+        self.paymentController = paymentController
+        super.init(
+            delegate: delegate,
+            session: session,
+            methodType: methodType,
+            apiClient: apiClient
+        )
+    }
+    
+    /// Launch Apple Pay sheet to confirm the payment intent
+    func startPayment(cancelPaymentOnDismiss: Bool = true) throws {
+        try AWXApplePayProvider.validate(paymentMethodType: paymentMethodType, session: unifiedSession)
+        paymentState = .notPresented
+        didHandlePresentationFail = false
+        didDismissWhilePending = false
+        self.cancelPaymentOnDismiss = cancelPaymentOnDismiss
+        
+        let request = try unifiedSession.makePaymentRequestOrError()
+        let controller = paymentController.init(paymentRequest: request)
+        controller.delegate = self
+        
+        Task { @MainActor in
+            let presented = await controller.present()
+            guard presented else {
+                handlePresentationFail()
+                return
+            }
+            delegate?.providerDidStartRequest(self)
+            
+            // Log risk event
+            RiskLogger.log(.showApplePay, screen: .applePay)
+            debugLog("Show apple pay")
+            paymentState = .notStarted
+            AnalyticsLogger.log(
+                pageView: .applePaySheet,
+                extraInfo: [
+                    .supportedNetworks : unifiedSession.applePayOptions?.supportedNetworks ?? []
+                ]
+            )
+        }
+    }
+    
+    /// Handle a failure to present the Apple Pay sheet
+    private func handlePresentationFail() {
+        if !didHandlePresentationFail {
+            didHandlePresentationFail = true
+            let error = "Failed to present Apple Pay Controller.".asError()
+            AnalyticsLogger.log(errorName: "apple_pay_sheet", errorMessage: error.rawValue)
+            delegate?.provider(self, didCompleteWith: .failure, error: error)
+        }
+    }
+}
+
+// MARK: - PKPaymentAuthorizationControllerDelegate
+
+extension ApplePayProvider: PKPaymentAuthorizationControllerDelegate {
+    
+    func paymentAuthorizationController(_ controller: PKPaymentAuthorizationController,
+                                        didAuthorizePayment payment: PKPayment) async -> PKPaymentAuthorizationResult {
+        await confirmIntent(payment: payment)
+    }
+    
+    @MainActor func confirmIntent(payment: PKPayment) async -> PKPaymentAuthorizationResult {
```

Comment:
This can be moved to the main body of the class and also marked as `private` since it isn't a requirement of the PKPaymentAuthorizationControllerDelegate protocol but a custom helper method.